### PR TITLE
[ROCm] Updating Dockerfile.rocm / CI scripts to use ROCm 3.3

### PR DESCRIPTION
**Pull request by deven-amd**
/cc @whchung @cheshire @chsigg @nvining-work

### CODE REVIEW

File: tensorflow/tools/ci_build/Dockerfile.rocm [hunk 1]
```diff
@@ -71,7 +71,20 @@ ENV PATH="$ROCM_PATH/bin:${PATH}"
...
+# The env vars being set via --action_env in .bazelrc and .tf_configure.bazelrc files are sometimes
+# not getting set in the build command being spawned by bazel (in theory this should not happen)
+# As a consequence ROCM_PATH is sometimes not set for the hipcc commands.
+# When hipcc incokes hcc, it specifies $ROCM_PATH/.../include dirs via the `-isystem` options
+# If ROCM_PATH is not set, it defaults to /opt/rocm, and as a consequence a dependency is generated on the
+# header files included within `/opt/rocm`, which then leads to bazel dependency errors
+# Explicitly creating the $ROCM_PATH/.info/version allows ROCM path to be set correrctly, even when ROCM_PATH
+# is not explicitly set, and thus avoids the eventual bazel dependency error.
+# The bazel bug needs to be root-caused and addressed, but that is out of our control and may take a long time
+# to come to fruition, so implementing the workaround to make do till then
```
Comment (chsigg) [inline tensorflow/tools/ci_build/Dockerfile.rocm:86]: Could you please reference the bazel issue that you filed here?

Comment (deven-amd) [reply tensorflow/tools/ci_build/Dockerfile.rocm:86]: @chsigg I have not filed a bazel issue for this, but I can. The only way I know to reproduce this issue, is to do a `--config=rocm` build with the `bazel -s` option, save the output to file, and then scan through the subcommand invocations and check whether all the `action_env`s specified on the bazel command were passes through correctly or not. Will that suffice?

Comment (deven-amd) [reply tensorflow/tools/ci_build/Dockerfile.rocm:86]: filed issue https://github.com/bazelbuild/bazel/issues/11163, and updated the comment to mention it

Comment (deven-amd) [reply tensorflow/tools/ci_build/Dockerfile.rocm:86]: @chsigg gentle ping

Comment (deven-amd) [reply tensorflow/tools/ci_build/Dockerfile.rocm:86]: @chsigg gentle ping

**Review by chsigg — APPROVED** [review]

**Pull request merged**